packages/redux-query-react/src/hooks/use-memoized-query-config.js

```diff
@@ -26,13 +26,15 @@ const useMemoizedQueryConfig = (
 
     if (
       queryKey !== previousQueryKey.current ||
-      headersChanged([queryConfig], [previousQueryConfig.current])
+      (queryConfig != null &&
+        previousQueryConfig.current != null &&
+        headersChanged([queryConfig], [previousQueryConfig.current]))
     ) {
       previousQueryKey.current = queryKey;
       previousQueryConfig.current = queryConfig;
       setQueryConfig(providedQueryConfig ? transform(providedQueryConfig) : null);
     }
-  }, [providedQueryConfig, transform]);
+  }, [providedQueryConfig, queryConfig, transform]);
```

Comment:
From my understanding, it is better to use `providedQueryConfig` to replace `queryConfig` to avoid unnecessary React.useEffect. And we can then remove  `queryConfig` from our dependencies.

Reply:
interesting, I was going by the lint rule that we use for the purpose for this.

Reply:
actually, that is exactly what is happening, it is linting this due to that. should I add this as an exception?